tools/oeedger8r/src/Sources.ml

```diff
@@ -763,6 +763,8 @@ let generate_trusted (ec : enclave_content) (ep : Intel.Util.edger8r_params) =
   [
     sprintf "#include \"%s_t.h\"" ec.file_shortnm;
     "";
+    "#define OE_NEED_STDC_NAMES";
```

Comment:
@jhand2 Where did we finally land on removing the use of this definition to require that generated stubs explicitly reference the `oe_*` prefixed versions of the functions? I see it's removed in enclave/core/tee_t_wrapper.c but it persists in the other wrappers.

Reply:
So @anakrish didn't want to use oe_* names because the interface isn't well-defined. I am looking into how I can just rely on the compiler to fill in all the functionality we need (which is already done today for wcslen for example).

I worry that this approach is too dependent on the compiler implementation though.

The other option is to define all the functions we need to include/openenclave/edger8r/common.h as inline functions. I don't love this either because code duplication but at least it has well-defined behavior.

Reply:
@anakrish  One thing about corelibc though. We actually do make guarantees about its stability. It is part of the public interface, so I actually think edger8r depending on corelibc functions seems ok from that perspective.

Reply:
Here is our  current documentation:
https://openenclave.io/apidocs/v0.8/files.html

That we don't even list any of the files in corelibc would indicate that it is more of an implementation detail rather than a  well-defined public API.

Reply:
My recommendation would be to add any functions that the edger8r needs to [include/edger8r/common.h](https://github.com/openenclave/openenclave/blob/71edd2cf92f52017410d5e3262628b962067560a/include/openenclave/edger8r/common.h)
and tweak the generator to use those functions. Say `oe_edl_malloc` instead of `malloc`, `oe_edl_strlen` instead of `strlen`.

This will make edger8r generated code more self-contained, while removing dependencies on libc. It also makes the requirements of the generated code very clear.

Reply:
@jhand2 Just now saw your earlier comment about adding functions to `edger8r/common.h`. So yes, I agree with you. Whether we inline or not is secondary; but the includes in `edger8r` present a well defined interface for what every TEE must implement to allow edger8r generated code to work.

For example, here is how OPTEE implements the edl functions [enclave/core/optee/hostcalls.c](https://github.com/openenclave/openenclave/blob/71edd2cf92f52017410d5e3262628b962067560a/enclave/core/optee/hostcalls.c)

Reply:
I'm not sure I fully follow the discussion presented here, but I don't have specific objections to the various options presented here. It sounds like there's an agreement to use edger8r/common.h as the consolidation point for definitions that edger8r stubs have on the OE implemented libc implementations; that sounds like it probably deserves its own issue and PR.

Reply:
I guess I don't understand why `oe_edl_strlen` is better than `oe_strlen`. Do we expect code generated by EDL to work with all versions of OE no matter which version it was generated with? That seems like a big promise.

I think the freestanding libc is more complicated and less stable than we're letting on due to reasons I detailed here: https://github.com/openenclave/openenclave/pull/2767#issuecomment-606729916

Reply:
As an aside, the issue with OE_NEED_STDC_NAMES is name collision. If this is defined and you want to use full (musl) libc there are likely be name collision.